android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SubscribableSensorKernelService.kt

```diff
@@ -0,0 +1,150 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package host.exp.exponent.kernel.services.sensors
+
+import android.content.Context
+import android.hardware.SensorEvent
+import host.exp.exponent.kernel.ExperienceKey
+import java.lang.ref.WeakReference
+import java.util.*
+
+private var DEFAULT_UPDATE_INTERVAL = 100
+
+abstract class SubscribableSensorKernelService internal constructor(reactContext: Context) : BaseSensorKernelService(reactContext) {
+  private val experienceScopeKeyListenersCountMap: MutableMap<String?, Int> = mutableMapOf()
```

Comment:
The null keys look sketchy but I see the comments below. At least Kotlin makes this code explicitly sketchy

Reply:
Yeah, I tried to remove the nullable key, but it'll take a follow-up PR adding a null check somewhere at the callsites to fix.